test/matrix/utility.js

```diff
@@ -176,4 +176,49 @@ describe('utility methods', function () {
         matrix2 = new Matrix([[2, 1, 3], [7, 1, 1], [6, 2, 7]]);
         matrix.strassen3x3(matrix2).to2DArray().should.eql([[38, 8, 17], [33, 12, 36], [35, 12, 37]]);
     });
+
+    it('pseudoinverse', function () {
+        // Actual values calculated by the Numpy library
+
+        var matrix = new Matrix([[2, 4], [7, 1]]);
+        var result = matrix.pseudoInverse().to2DArray();
+
+        result[0][0].should.approximately(-0.03846153846153843, 1e-1);
+        result[0][1].should.approximately(0.15384615384615374, 1e-1);
+        result[1][0].should.approximately(0.2692307692307691, 1e-1);
+        result[1][1].should.approximately(-0.076923076923077, 1e-1);
+
+        matrix = new Matrix([[4, 7], [2, 6]]);
+        result = matrix.pseudoInverse().to2DArray();
+        result[0][0].should.approximately(0.6, 1e-1);
+        result[0][1].should.approximately(-0.7, 1e-1);
+        result[1][0].should.approximately(-0.2, 1e-1);
+        result[1][1].should.approximately(0.4, 1e-1);
+
+        matrix = new Matrix([[3, 0, 2], [2, 0, -2], [0, 1, 1]]);
+        result = matrix.pseudoInverse().to2DArray();
+
+        result[0][0].should.approximately(0.2, 1e-1);
+        result[0][1].should.approximately(0.2, 1e-1);
+        result[0][2].should.approximately(0, 1e-1);
+
+        result[1][0].should.approximately(-0.2, 1e-1);
+        result[1][1].should.approximately(0.3, 1e-1);
+        result[1][2].should.approximately(1, 1e-1);
+
+        result[2][0].should.approximately(0.2, 1e-1);
+        result[2][1].should.approximately(-0.3, 1e-1);
+        result[2][2].should.approximately(0, 1e-1);
+
+        matrix = new Matrix([[1, 0, 1], [2, 4, 5]]);
+        result = matrix.pseudoInverse().to2DArray();
+
+        result[0][0].should.approximately(0.75609756, 1e-1);
+        result[0][1].should.approximately(-0.07317073, 1e-1);
+        result[1][0].should.approximately(-0.68292683, 1e-1);
+        result[1][1].should.approximately(0.19512195, 1e-1);
+        result[2][0].should.approximately(0.24390244, 1e-1);
+        result[2][1].should.approximately(0.07317073, 1e-1);
+
```

Comment:
Can you be less tolerant for errors? At least 1e-5 if possible.
Can you also add the example from Wolfram: http://reference.wolfram.com/language/ref/PseudoInverse.html